replica/table.cc

```diff
@@ -1286,26 +1286,41 @@ future<> table::perform_cleanup_compaction(compaction::owned_ranges_ptr sorted_o
 }
 
 unsigned table::estimate_pending_compactions() const {
-    return boost::accumulate(compaction_groups() | boost::adaptors::transformed([this] (const compaction_group_ptr& cg) {
-        return _compaction_strategy.estimated_pending_compactions(cg->as_table_state());
+    return boost::accumulate(compaction_groups() | boost::adaptors::transformed([] (const compaction_group_ptr& cg) {
+        return cg->compaction_strategy().estimated_pending_compactions(cg->as_table_state());
     }), unsigned(0));
 }
 
+void compaction_group::set_compaction_strategy(sstables::compaction_strategy&& cs) noexcept {
+    _compaction_strategy = std::move(cs);
+}
+
+sstables::compaction_strategy& compaction_group::compaction_strategy() noexcept {
+    return _compaction_strategy;
+}
+
+const sstables::compaction_strategy& table::get_compaction_strategy() const {
+    return _compaction_groups.front()->compaction_strategy();
+}
```

Comment:
I think we either need to remove this method altogether -- and have everybody explicitly use a compaction-group to retrieve the compaction strategy -- or separate compaction strategy into a separate state stored in the compaction group and a pure strategy possibly stored in the table.
Doesn't have to be done now, just raising the idea.